src/components/Footer.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { TodosContext } from '../stores/TodosContext';
+import { FilterParams } from '../types/FilterParams';
+
+type Props = {
+  filterBy: FilterParams;
+  setFilterBy: (v: FilterParams) => void;
+};
+
+export const Footer: React.FC<Props> = ({ filterBy, setFilterBy }) => {
+  const {
+    deleteTodo,
+    todos,
+    unCompletedTodos,
+    completedTodos,
+  } = useContext(TodosContext);
+
+  const todosCounter = unCompletedTodos.length;
+
+  const handleAllClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    setFilterBy(FilterParams.All);
+  };
+
+  const handleActiveClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    setFilterBy(FilterParams.Active);
+  };
+
+  const handleCompletedClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    setFilterBy(FilterParams.Completed);
+  };
+
+  function handleDeleteCompleted() {
+    todos
+      .filter(({ completed }) => completed)
+      .forEach(({ id }) => deleteTodo(id));
+  }
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${todosCounter} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={
+            classNames('filter__link', {
+              selected: filterBy === FilterParams.All,
+            })
+          }
+          data-cy="FilterLinkAll"
+          onClick={handleAllClick}
```

Comment:
```suggestion
          onClick={handleFilter(FilterParams.All)}
```
same for other filter links